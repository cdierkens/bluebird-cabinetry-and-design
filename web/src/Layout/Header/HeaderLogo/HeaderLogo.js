import { Link } from "gatsby";
import { PropTypes } from "prop-types";
import React from "react";
import { useScrollPosition } from "../../../hooks";
import logo from "../../../logo.png";
import styles from "./HeaderLogo.module.css";

const HeaderLogo = ({ className = "" }) => {
  const position = useScrollPosition();

  return (
    <div className={`${styles.HeaderLogo} ${className}`}>
      <Link to="/">
        <img
          className={`${styles.Logo} ${
            position.y > 200 ? styles.LogoTransform : ""
          }`}
          src={logo}
          alt="Bluebird Logo"
        />
      </Link>
    </div>
  );
};

HeaderLogo.propTypes = {
  className: PropTypes.string,
};

export default HeaderLogo;
