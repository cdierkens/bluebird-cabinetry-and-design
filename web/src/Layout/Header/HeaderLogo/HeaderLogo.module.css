.HeaderLogo {
  @apply relative;
}

.Logo {
  @apply bg-blue-dark rounded-b-lg px-3 w-48 absolute z-10 transform origin-top transition duration-500 scale-100;
}

.LogoTransform {
  @apply scale-60;
}
