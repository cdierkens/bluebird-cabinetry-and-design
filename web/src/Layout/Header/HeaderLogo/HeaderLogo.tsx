import clsx from "clsx";
import { Link } from "gatsby";
import React from "react";
import { useScrollPosition } from "../../../hooks";
import logo from "../../../images/logo.png";
import * as styles from "./HeaderLogo.module.css";

interface HeaderLogoProps {
  className?: string;
}

const HeaderLogo: React.FC<React.PropsWithChildren<HeaderLogoProps>> = ({ className }) => {
  const position = useScrollPosition();

  return (
    <div className={clsx(styles.HeaderLogo, className)}>
      <Link to="/">
        <img
          className={clsx(styles.Logo, {
            [styles.LogoTransform]: position?.y && position.y > 200,
          })}
          src={logo}
          alt="Bluebird Logo"
        />
      </Link>
    </div>
  );
};

export default HeaderLogo;
