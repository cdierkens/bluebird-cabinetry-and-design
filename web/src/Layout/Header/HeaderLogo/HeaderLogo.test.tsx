import { render } from "@testing-library/react";
import React from "react";
import HeaderLogo from "./HeaderLogo";

it("matches snapshot", () => {
  const { asFragment } = render(<HeaderLogo />);

  expect(asFragment()).toMatchSnapshot();
});
