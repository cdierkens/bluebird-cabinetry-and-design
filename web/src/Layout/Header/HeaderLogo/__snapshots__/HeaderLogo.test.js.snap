// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="HeaderLogo "
  >
    <a
      href="/"
    >
      <img
        alt="Bluebird Logo"
        class="Logo "
        src="test-file-stub"
      />
    </a>
  </div>
</DocumentFragment>
`;
